### PR TITLE
[internal/comparetest] add golden functions - readLogs, writeLogs, compareLogs

### PR DESCRIPTION
**Pull request by jackycsl**
**Description:** <Describe what has changed.>

Extend [comparetest](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/comparetest) package to have readLogs, writeLogs, compareLogs

**Link to tracking Issue:** <Issue number if applicable>
#10896

**Testing:** <Describe what testing was performed and which tests were added.>
Add unit testing for readLogs, writeLogs, and compareLogs

**Documentation:** <Describe the documentation added.>
Add typical usage to README

### CODE REVIEW

File: internal/comparetest/compare.go [hunk 1]
```diff
@@ -267,3 +268,161 @@ func numberTypeToString(t pmetric.NumberDataPointValueType) string {
...
+
+// CompareLogs compares each part of two given Logs and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareLogs(expected, actual plog.Logs) error {
+	if expected.ResourceLogs().Len() != actual.ResourceLogs().Len() {
+		return fmt.Errorf("amount of ResourceLogs between Logs are not equal (expected: %d, actual: %d)",
+			expected.ResourceLogs().Len(),
+			actual.ResourceLogs().Len())
+	}
+
```
Comment (dmitryax) [inline internal/comparetest/compare.go:280]: We should make copies of plog.Logs and sort all the slices as it's done for metrics. Right?

Comment (djaglowski) [reply internal/comparetest/compare.go:280]: @jackycsl, do you plan on iterating on this PR?

File: internal/comparetest/README.md [hunk 2]
```diff
@@ -27,18 +27,44 @@ func TestScraper(t *testing.T) {
...
+  return sink.LogRecordCount() > 0
+  }, 2*time.Second, 10*time.Millisecond)
+
+  err = alertRcvr.Shutdown(context.Background())
+  require.NoError(t, err)
+
+  logs := sink.AllLogs()[0]
+  expected, err := readLogs(filepath.Join("testdata", "logs", "expected.json"))
+  require.NoError(t, err)
+  require.NoError(t, compareLogs(expected, logs))
```
Comment (dmitryax) [inline internal/comparetest/README.md:52]:
```suggestion
  require.NoError(t, comparetest.CompareLogs(expected, logs))
```

File: internal/comparetest/README.md [hunk 3]
```diff
@@ -27,18 +27,44 @@ func TestScraper(t *testing.T) {
...
+
+  sink := &consumertest.LogsSink{}
+  alertRcvr := newLogsReceiver(cfg, zap.NewNop(), sink)
+  alertRcvr.client = defaultMockClient()
+
+  err := alertRcvr.Start(context.Background(), componenttest.NewNopHost())
+  require.NoError(t, err)
+
+  require.Eventually(t, func() bool {
+  return sink.LogRecordCount() > 0
```
Comment (dmitryax) [inline internal/comparetest/README.md:43]:
```suggestion
    return sink.LogRecordCount() > 0
```

**Review by dmitryax — COMMENTED** [review]
Couple nits, otherwise LGTM

File: internal/comparetest/compare.go [hunk 4]
```diff
@@ -267,3 +268,210 @@ func numberTypeToString(t pmetric.NumberDataPointValueType) string {
...
+			actual.TraceID())
+	}
+
+	if expected.SpanID() != actual.SpanID() {
+		return fmt.Errorf("log record SpanID doesn't match expected: %d, actual: %d",
+			expected.SpanID(),
+			actual.SpanID())
+	}
+
+	if !expected.Body().Equal(actual.Body()) {
```
Comment (dmitryax) [inline internal/comparetest/compare.go:470]: Map.Equal method is deprecated

```suggestion
	if expected.Body().AsRaw() != actual.Body().AsRaw() {
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**